packages/teleterm/src/sharedProcess/ptyHost/server/ptyEventsStreamHandler.ts

```diff
@@ -0,0 +1,91 @@
+import { ServerDuplexStream } from '@grpc/grpc-js';
+import {
+  PtyClientEvent,
+  PtyEventData,
+  PtyEventExit,
+  PtyEventOpen,
+  PtyEventResize,
+  PtyEventStart,
+  PtyServerEvent,
+} from '../v1/ptyHostService_pb';
+import { PtyProcess } from '../server/ptyProcess';
+
+export class PtyEventsStreamHandler {
+  private ptyId?: string;
+
+  constructor(
+    private readonly stream: ServerDuplexStream<PtyClientEvent, PtyServerEvent>,
+    private readonly ptyProcesses: Map<string, PtyProcess>
```

Comment:
> Unfortunately, there is no way to add any data when the channel is created (so we could set `ptyId` for it). To overcome it, we set `ptyId` in the first event which is sent (`start`), so _sever_ knows which `process` this channel belongs to.

It'd be a good idea to document this somewhere, perhaps in this constructor? It'd help explain why `PtyEventsStreamHandler` cannot take `ptyId` as a constructor argument.

Reply:
Good idea, done